app/auth/Authenticated.scala

```diff
@@ -35,7 +37,16 @@ trait Authenticated {
 
   val auth: AuthConnector
 
-  def authenticated(f: => AuthenticationResult => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
+  def authenticated(f: => Future[Result])(implicit hc: HeaderCarrier): Future[Result] =
+    auth.getCurrentAuthority().flatMap { authority =>
+      Logger.debug(s"Got authority = $authority")
+      mapToAuthResult(authority) match {
+        case NotLoggedIn => Unauthorized
+        case _ => f
+      }
+    }
+
+  def authenticatedCustom(f: => AuthenticationResult => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
```

Comment:
not sure what to call this one, I'm open to suggestions...

Basically in this function you need to pass in a function which you must specify how to deal with the authentication results. Whereas the other function it has default behaviours.

The custom function is solely used for the throttling at the moment so I could call it something along the lines of throttleAuthentication?